src/devices/Tca9548a/Tca9548A.cs

```diff
@@ -0,0 +1,167 @@
+        public Tca9548A(I2cBus i2CBus, byte address = DefaultI2cAddress, bool shouldDispose = true)
+        {
+            _i2cBus = i2CBus;
+            _i2CDevice = _i2cBus.CreateDevice(address);
+            _shouldDispose = shouldDispose;
+        }
+
+        /// <summary>
+        /// Creates an I2c Device for selected sensor on Multiplexer
+        /// </summary>
+        /// <param name="tcaChannel">TCA Channel on which Device Exists</param>
+        /// <param name="address">I2c address of the device</param>
+        /// <returns></returns>
+        public I2cDevice CreateDeviceOnChannel(Channels tcaChannel, int address)
+        {
+            return new Tca9548ADevice(this, tcaChannel, address);
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select [2^channel number (0-7)]</param>
+        public void SelectChannel(byte multiplexChannel)
+        {
+            try
+            {
+                _i2CDevice.WriteByte(multiplexChannel);
+                _i2CDevice.ReadByte();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Unable to select ({_i2CDevice.ConnectionSettings.BusId} ,{_i2CDevice.ConnectionSettings.DeviceAddress}, {multiplexChannel}): {e}");
+            }
+        }
+
+        /// <summary>
+        /// Select a multiplex channel
+        /// </summary>
+        /// <param name="multiplexChannel">The channel to select</param>
+        public void SelectChannel(Channels multiplexChannel)
+        {
+            SelectChannel(Convert.ToByte(multiplexChannel));
+        }
+
+        /// <summary>
+        /// Try getting the selected channel on Multiplexer
+        /// </summary>
+        /// <param name="selectedChannel"> selected Multiplexer Channel</param>
+        /// <returns> true if able to retrieve selected channel</returns>
+        public bool TryGetSelectedChannel(out Channels selectedChannel)
+        {
+            try
+            {
+                var channel = _i2CDevice.ReadByte();
+                if (Enum.IsDefined(typeof(Channels), channel))
+                {
+                    selectedChannel = (Channels)channel;
+                    return true;
+                }
+                else
+                {
+                    selectedChannel = Channels.Channel0;
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                selectedChannel = Channels.Channel0;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns all the connected devices on all channels of Multiplexer
+        /// </summary>
+        /// <returns> Dictionary having Keys as Channels and Values as IEnumerable integer type</returns>
+        public Dictionary<Channels, IEnumerable<int>> ScanAllChannelsForDeviceAddress()
```

Comment:
Once you add CreateI2cBus I believe this won't be needed since you will get this for free, is my understanding correct?